### PR TITLE
Catch UDF compiler exceptions and fallback to CPU

### LINKED ISSUE #3404: [FEA] UDF compiler crashes should not prevent query from running
**Is your feature request related to a problem? Please describe.**
#3369 is an example of a crash in the UDF compiler that prevented the query from running.  Since we can fallback to the CPU for UDFs, it seems unnecessary to fail the entire query if a particular UDF causes a crash in the UDF compiler.

**Describe the solution you'd like**
The plugin should catch exceptions thrown by the UDF compiler during planning and have the UDF fallback to the CPU if the UDF compiler throws an exception processing the UDF.

### PR DESCRIPTION
**Pull request by jlowe**
Fixes #3404 

Updates the UDF compiler to catch other types of exceptions besides the typical `SparkException` used to indicate a translation failure and fallback to the CPU in case of any exception.  Exceptions other than `SparkException` are logged as a warning.

### CODE REVIEW

**Comment by jlowe** [comment]
build

File: udf-compiler/src/main/scala/com/nvidia/spark/udf/GpuScalaUDF.scala [hunk 1]
```diff
@@ -50,11 +50,16 @@ case class GpuScalaUDFLogical(udf: ScalaUDF) extends ShimExpression with Logging
...
     } catch {
       case e: SparkException =>
-        logDebug("UDF compilation failure: " + e)
+        val udfName = udf.udfName.getOrElse("<unknown>")
+        logDebug(s"UDF $udfName compilation failure: $e")
         if (isTestEnabled) {
           throw e
         }
         udf
+      case e: Exception =>
```
Comment (gerashegalov) [inline udf-compiler/src/main/scala/com/nvidia/spark/udf/GpuScalaUDF.scala:59]: prefer the idiom `case NonFatal(e) =>`

**Comment by jlowe** [comment]
build

**Review by gerashegalov — APPROVED** [review]

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**